src/devices/AndroidDevice.tsx

```diff
@@ -163,4 +164,19 @@ export default class AndroidDevice extends BaseDevice {
     this.recordingProcess = undefined;
     return destination;
   }
+
+  async getProcessName(pid: number): Promise<string> {
+    return await this.adb
```

Comment:
I tried running Flipper with your changes, the performance of the app took a hit. It became unresponsive. I reckon, it is because of the `await` here. We wait for the `getProcessName` to finish and then we call `processEntry`. I will request you consider going for non blocking approach and avoid await.

Reply:
@priteshrnandgaonkar I have another approach in mind. But, I felt the way I have done is more intuitive for users. BTW, how did you monitor the performance? I will use the same once I do modifications

Reply:
I didn't use any tool. I just ran Flipper and tried to use it normally, it was unresponsive.

Reply:
The problem here is that this function is being called for every single log line, so when hundreds are coming in, it's starting up ps and grep processes many times per second. This isn't necessary - we don't need to recheck the pid again for every single log.

A better approach would be to memoize and reuse the process name for each device + pid, bearing in mind that they will occasionally change, so it should not cache forever.

In addition, if the adkbit promise takes a while to return, many log listener process entry calls are getting queued up until it resolves. I'm not sure if this is a problem in practice. I'd start by fixing the above one - I suspect thats the main issue.

Reply:
The reason why it is being called for every single log line is, at some point some process gets killed and the same `pid` can be assigned it to different process. So, the complexity here to memoize is, I can't make pid or process name as key. I tried to use `Select` component and once we select a process from that list I tried to filter the rows. Still, the application is becoming unresponsive after a minute or two. If we can get the process name from `adbkit` itself, I think we can gain better performance. @priteshrnandgaonkar @jknoxville Can you suggest?

Reply:
Filtering by selected application is a killer feature. I would start using Flipper again once this is finished.

> If we can get the process name from adbkit itself, I think we can gain better performance.

@jknoxville what do you think? Is that a good approach?

Personally, I don't know this part of the project.